diff --git a/node_modules/@cypress/request/request.js b/node_modules/@cypress/request/request.js
index 53e6177..0dee7fb 100644
--- a/node_modules/@cypress/request/request.js
+++ b/node_modules/@cypress/request/request.js
@@ -1069,9 +1069,9 @@ Request.prototype.onRequestResponse = function (response) {
     responseContent.once('end', function (chunk) {
       self.emit('end', chunk)
     })
-    responseContent.on('error', function (error) {
-      self.emit('error', error)
-    })
+    // responseContent.on('error', function (error) {
+    //   self.emit('error', error)
+    // })
     responseContent.on('close', function () { self.emit('close') })
 
     if (self.callback) {
