diff --git a/node_modules/@types/jquery.scrollto/index.d.ts b/node_modules/@types/jquery.scrollto/index.d.ts
index 25110fb..d23c9ce 100644
--- a/node_modules/@types/jquery.scrollto/index.d.ts
+++ b/node_modules/@types/jquery.scrollto/index.d.ts
@@ -1,6 +1,6 @@
 /// <reference types="jquery"/>
 
-interface ScrollToOptions {
+interface JQueryScrollToOptions<TElement> extends JQuery.EffectsOptions<TElement> {
     /**
      * Which axis must be scrolled, use 'x', 'y', 'xy' or 'yx'.
      */
@@ -42,7 +42,7 @@ interface ScrollToOptions {
     onAfterFirst?: (() => void) | undefined;
 }
 
-interface JQuery {
+interface JQuery<TElement> {
     /**
      * Scroll the matched elements
      */
@@ -54,7 +54,7 @@ interface JQuery {
          * @param duration The OVERALL length of the animation
          * @param settings Set of settings.
          */
-        (target: any, duration?: number, settings?: ScrollToOptions): JQuery;
+        (target: any, duration?: number, settings?: JQueryScrollToOptions<TElement>): JQuery;
         /**
          * Scroll the matched elements
          *
@@ -70,7 +70,7 @@ interface JQuery {
          * @param settings Set of settings.
          * @param onAfter The onAfter callback.
          */
-        (target: any, settings: ScrollToOptions, onAfter?: Function): JQuery;
+        (target: any, settings: JQueryScrollToOptions<TElement>, onAfter?: Function): JQuery;
     };
 }
 
@@ -86,7 +86,7 @@ interface JQueryStatic {
          * @param duration The OVERALL length of the animation
          * @param settings Set of settings.
          */
-        (target: any, duration?: number, settings?: ScrollToOptions): JQuery;
+        (target: any, duration?: number, settings?: JQueryScrollToOptions<any>): JQuery;
         /**
          * Scroll window
          *
@@ -102,6 +102,6 @@ interface JQueryStatic {
          * @param settings Set of settings.
          * @param onAfter The onAfter callback.
          */
-        (target: any, settings: ScrollToOptions, onAfter?: Function): JQuery;
+        (target: any, settings: JQueryScrollToOptions<any>, onAfter?: Function): JQuery;
     };
 }
