diff --git a/node_modules/bytenode/lib/index.js b/node_modules/bytenode/lib/index.js
index cdd98cc..cc6b8fa 100644
--- a/node_modules/bytenode/lib/index.js
+++ b/node_modules/bytenode/lib/index.js
@@ -327,10 +327,7 @@ const addLoaderFile = function (fileToLoad, loaderFilename) {
 };
 
 const loaderCode = function (targetPath) {
-  return `
-    require('bytenode');
-    require('${targetPath}');
-  `;
+  return `require('bytenode');require('${targetPath}');`;
 };
 
 global.bytenode = {
