diff --git a/node_modules/electron-context-menu/index.d.ts b/node_modules/electron-context-menu/index.d.ts
index 95ade1c..03293e3 100644
--- a/node_modules/electron-context-menu/index.d.ts
+++ b/node_modules/electron-context-menu/index.d.ts
@@ -1,13 +1,15 @@
 import {
 	BrowserWindow,
 	BrowserView,
-	WebviewTag,
 	ContextMenuParams,
 	MenuItemConstructorOptions,
 	Event as ElectronEvent,
-	WebContents
+	WebContents,
 } from 'electron';
 
+// TODO: remove this patch once electron-context-menu is fixed
+type WebviewTag = Electron.WebviewTag
+
 declare namespace contextMenu {
 	interface Labels {
 		/**
