diff --git a/node_modules/make-empty-github-commit/src/index.js b/node_modules/make-empty-github-commit/src/index.js
index db10861..f928cc6 100644
--- a/node_modules/make-empty-github-commit/src/index.js
+++ b/node_modules/make-empty-github-commit/src/index.js
@@ -1,5 +1,6 @@
 var { Octokit } = require('@octokit/rest')
 var debug = require('debug')('make-empty-github-commit')
+var { createAppAuth } = require('@octokit/auth-app')
 
 var getReferenceCommit = async function (data) {
   const res = await data.octokit.rest.git.getRef({
@@ -56,6 +57,7 @@ function emptyGitHubCommit (opts) {
   }
   var data = {}
   data.octokit = new Octokit({
+    authStrategy: createAppAuth,
     auth: opts.token
   })
   data.owner = opts.owner
