diff --git a/node_modules/snap-shot-it/src/index.js b/node_modules/snap-shot-it/src/index.js
index 0fb68da..154f09d 100644
--- a/node_modules/snap-shot-it/src/index.js
+++ b/node_modules/snap-shot-it/src/index.js
@@ -319,6 +319,11 @@ function snapshot (value) {
       // the code is duplicate from above to get just the key collision error
       const info = R.assoc('key', e.key, snapshotInfo)
       info.allowDuplicate = Boolean(snapshotOptions.allowSharedSnapshot)
+
+      if (info.allowDuplicate) {
+        return
+      }
+      
       debug('current snapshot info %o', info)
 
       const prevInfo = findExistingSnapshotKey(info)
