diff --git a/node_modules/ts-node/dist/index.js b/node_modules/ts-node/dist/index.js
index c03afbf..56c45bb 100644
--- a/node_modules/ts-node/dist/index.js
+++ b/node_modules/ts-node/dist/index.js
@@ -94,7 +94,7 @@ exports.DEFAULTS = {
 /**
  * TypeScript diagnostics error.
  */
-class TSError extends make_error_1.BaseError {
+class TSError extends Error {
     constructor(diagnosticText, diagnosticCodes, diagnostics = []) {
         super(`⨯ Unable to compile TypeScript:\n${diagnosticText}`);
         this.diagnosticCodes = diagnosticCodes;
@@ -110,12 +110,6 @@ class TSError extends make_error_1.BaseError {
             value: diagnostics,
         });
     }
-    /**
-     * @internal
-     */
-    [exports.INSPECT_CUSTOM]() {
-        return this.diagnosticText;
-    }
 }
 exports.TSError = TSError;
 const TS_NODE_SERVICE_BRAND = Symbol('TS_NODE_SERVICE_BRAND');
