diff --git a/node_modules/vite/dist/node/chunks/dep-c6273c7a.js b/node_modules/vite/dist/node/chunks/dep-c6273c7a.js
index 22e37a1..ee75032 100644
--- a/node_modules/vite/dist/node/chunks/dep-c6273c7a.js
+++ b/node_modules/vite/dist/node/chunks/dep-c6273c7a.js
@@ -62830,7 +62830,8 @@ async function bundleConfigFile(fileName, isESM) {
                             if (path$n.relative(idPkgDir, fileName).startsWith('..')) {
                                 return {
                                     // normalize actual import after bundled as a single vite config
-                                    path: idFsPath,
+                                    // TODO: remove this patch once Vite 3.0.5 is released.
+                                    path: pathToFileURL(idFsPath).href,
                                     external: true
                                 };
                             }
