diff --git a/node_modules/axios/lib/platform/common/utils.js b/node_modules/axios/lib/platform/common/utils.js
index 52a3186..08b92cc 100644
--- a/node_modules/axios/lib/platform/common/utils.js
+++ b/node_modules/axios/lib/platform/common/utils.js
@@ -1,6 +1,11 @@
-const hasBrowserEnv = typeof window !== 'undefined' && typeof document !== 'undefined';
+/**
+ * patched due to how v8 snapshots work - axios was incorrectly determining that
+ * it was running in a browser context, leading to runtime errors.
+ */
+
+const hasBrowserEnv = false;
 
-const _navigator = typeof navigator === 'object' && navigator || undefined;
+const _navigator = undefined;
 
 /**
  * Determine if we're running in a standard browser environment
