diff --git a/node_modules/foxdriver/build/logger.js b/node_modules/foxdriver/build/logger.js
index 80e315d..c160f30 100644
--- a/node_modules/foxdriver/build/logger.js
+++ b/node_modules/foxdriver/build/logger.js
@@ -7,7 +7,7 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.default = Logger;
 
-var _npmlog = _interopRequireDefault(require("npmlog"));
+const debug = require('debug')('foxdriver')
 
 var _package = _interopRequireDefault(require("../package.json"));
 
@@ -16,10 +16,6 @@ var _package = _interopRequireDefault(require("../package.json"));
  */
 const NPM_LEVELS = ['silly', 'verbose', 'debug', 'info', 'http', 'warn', 'error', 'chrome', 'firefox'];
 
-_npmlog.default.addLevel('debug', 1000, {
-  fg: 'blue',
-  bg: 'black'
-}, 'dbug');
 
 function Logger(component) {
   const wrappedLogger = {};
@@ -30,10 +26,8 @@ function Logger(component) {
 
   Object.defineProperty(wrappedLogger, 'level', {
     get: () => {
-      return _npmlog.default.level;
     },
     set: newValue => {
-      _npmlog.default.level = newValue;
     },
     enumerable: true,
     configurable: true
@@ -44,8 +38,8 @@ function Logger(component) {
 
   for (let level of NPM_LEVELS) {
     wrappedLogger[level] = (...args) => {
-      if (!process.env.DEBUG) return;
-      return _npmlog.default[level](prefix, ...args);
+      // @see https://github.com/cypress-io/cypress/issues/7723
+      debug(prefix, ...args)
     };
   }
 
