diff --git a/node_modules/http-proxy/.auto-changelog b/node_modules/http-proxy/.auto-changelog
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/.gitattributes b/node_modules/http-proxy/.gitattributes
deleted file mode 100644
index 1a6bd45..0000000
--- a/node_modules/http-proxy/.gitattributes
+++ /dev/null
@@ -1 +0,0 @@
-package-lock.json binary
diff --git a/node_modules/http-proxy/CHANGELOG.md b/node_modules/http-proxy/CHANGELOG.md
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/CODE_OF_CONDUCT.md b/node_modules/http-proxy/CODE_OF_CONDUCT.md
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/LICENSE b/node_modules/http-proxy/LICENSE
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/README.md b/node_modules/http-proxy/README.md
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/codecov.yml b/node_modules/http-proxy/codecov.yml
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/index.js b/node_modules/http-proxy/index.js
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/lib/http-proxy.js b/node_modules/http-proxy/lib/http-proxy.js
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/lib/http-proxy/common.js b/node_modules/http-proxy/lib/http-proxy/common.js
old mode 100644
new mode 100755
index 6513e81..486d4c8
--- a/node_modules/http-proxy/lib/http-proxy/common.js
+++ b/node_modules/http-proxy/lib/http-proxy/common.js
@@ -1,6 +1,5 @@
 var common   = exports,
     url      = require('url'),
-    extend   = require('util')._extend,
     required = require('requires-port');
 
 var upgradeHeader = /(^|,)\s*upgrade\s*($|,)/i,
@@ -40,10 +39,10 @@ common.setupOutgoing = function(outgoing, options, req, forward) {
   );
 
   outgoing.method = options.method || req.method;
-  outgoing.headers = extend({}, req.headers);
+  outgoing.headers = Object.assign({}, req.headers);
 
   if (options.headers){
-    extend(outgoing.headers, options.headers);
+    Object.assign(outgoing.headers, options.headers);
   }
 
   if (options.auth) {
diff --git a/node_modules/http-proxy/lib/http-proxy/index.js b/node_modules/http-proxy/lib/http-proxy/index.js
old mode 100644
new mode 100755
index 977a4b3..88b2d0f
--- a/node_modules/http-proxy/lib/http-proxy/index.js
+++ b/node_modules/http-proxy/lib/http-proxy/index.js
@@ -1,5 +1,4 @@
 var httpProxy = module.exports,
-    extend    = require('util')._extend,
     parse_url = require('url').parse,
     EE3       = require('eventemitter3'),
     http      = require('http'),
@@ -47,9 +46,9 @@ function createRightProxy(type) {
         args[cntr] !== res
       ) {
         //Copy global options
-        requestOptions = extend({}, options);
+        requestOptions = Object.assign({}, options);
         //Overwrite with request options
-        extend(requestOptions, args[cntr]);
+        Object.assign(requestOptions, args[cntr]);
 
         cntr--;
       }
diff --git a/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.js b/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.js
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/lib/http-proxy/passes/web-outgoing.js b/node_modules/http-proxy/lib/http-proxy/passes/web-outgoing.js
old mode 100644
new mode 100755
diff --git a/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js b/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
old mode 100644
new mode 100755
index 270f23f..feac117
--- a/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
+++ b/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
@@ -111,7 +111,7 @@ module.exports = {
     proxyReq.on('error', onOutgoingError);
     proxyReq.on('response', function (res) {
       // if upgrade event isn't going to happen, close the socket
-      if (!res.upgrade) {
+      if (!res.upgrade && socket.readyState === "open" && !socket.destroyed) {
         socket.write(createHttpHeader('HTTP/' + res.httpVersion + ' ' + res.statusCode + ' ' + res.statusMessage, res.headers));
         res.pipe(socket);
       }
diff --git a/node_modules/http-proxy/renovate.json b/node_modules/http-proxy/renovate.json
old mode 100644
new mode 100755
