diff --git a/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js b/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js
index 11dab90..d00a8f1 100644
--- a/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js
+++ b/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js
@@ -192,7 +192,7 @@ function withGlobal(_global) {
     isPresent.requestIdleCallback =
         _global.requestIdleCallback &&
         typeof _global.requestIdleCallback === "function";
-    isPresent.cancelIdleCallbackPresent =
+    isPresent.cancelIdleCallback =
         _global.cancelIdleCallback &&
         typeof _global.cancelIdleCallback === "function";
     isPresent.setImmediate =
