diff --git a/node_modules/merge-source-map/index.js b/node_modules/merge-source-map/index.js
index 2867fb7..efa44ed 100644
--- a/node_modules/merge-source-map/index.js
+++ b/node_modules/merge-source-map/index.js
@@ -47,7 +47,8 @@ function merge(oldMap, newMap) {
     })
   })
 
-  var consumers = [oldMapConsumer, newMapConsumer]
+  // fixes https://github.com/keik/merge-source-map/issues/6
+  var consumers = [newMapConsumer, oldMapConsumer]
   consumers.forEach(function(consumer) {
     consumer.sources.forEach(function(sourceFile) {
       mergedMapGenerator._sources.add(sourceFile)
